ffi_internals/src/consumer/consumer_enum.rs

```diff
@@ -56,10 +56,20 @@ extension {}: FFIArray {{
         format!(
             r#"
 public extension Optional where Wrapped == {} {{
-    func toRust() -> UnsafeMutablePointer<{}>? {{
+    func clone() -> UnsafeMutablePointer<{}>? {{
         switch self {{
         case let .some(value):
-            let v = value.toRust()
+            let v = value.clone()
+            return UnsafeMutablePointer(mutating: {}(true, v))
+        case .none:
+            return nil
+        }}
+    }}
+
+    func borrowReference() -> UnsafeMutablePointer<{}>? {{
+        switch self {{
+        case let .some(value):
+            let v = value.borrowReference()
```

Comment:
Just curious: if this syntax is bad, how long does it take to find out? Are there tests that catch that?

Reply:
Too long 😞 

There's nothing to catch it in `ffi_common` yet. There's a script in the mobile project for building `rust_monster`, and if there's an issue, you'll see compiler errors in the mobile project once that finishes.

To make this easier to maintain / expand, I'd really like to:
1. Pull all of this hardcoded Swift out into template files, and just pass a `ConsumerLanguage` enum to decide how we generate the consumer.
2. Test the resulting syntax by using something like https://github.com/sagiegurari/run_script to invoke a script that builds the generated Swift (or whatever else). That way the tests for the `consumer` module enforce compilable code without having to check in a client project (or doing any gross re-parsing of the consumer syntax, or hardcoding another layer to Swift-only).

Reply:
Nice. Yeah, seems like there should be at least something in the compile process that does some sort of checking. I know we briefly talked about this some time ago, but in my brain it sure would be awesome if there was a `quote`-like crate (or maybe there's a better paradigm for allowing you generate code in another language) that could generate swift code. testing could the be localized to that crate (ensuring it generates valid swift). easier said than done though, of course.